plugins/techdocs/src/reader/components/TechDocsReaderPage/TechDocsReaderPage.tsx

```diff
@@ -90,20 +55,26 @@ export const TechDocsReaderPage = ({
   const outlet = useOutlet();
   const entityName = entityRef ?? { kind, name, namespace };
 
+  // children is only going to be defined if no "element" prop is provided to the route
   if (!children) {
-    const childrenList = outlet ? Children.toArray(outlet.props.children) : [];
+    // the children of the outlet will always just be one component
+    // therefore we need to support either custom reader page OR addons (using our default reader page) and not both
```

Comment:
yes, if you compose your techdocs page in the app with the following (what we currently allow) and log out `outlet.props.children`: 
```tsx
<Route
      path="/docs/:namespace/:kind/:name/*"
      element={<TechDocsReaderPage />}
    >
       {techDocsPage}
      <TechDocsAddons>
        <ExampleHeader />
        <ExampleSubHeader />
        <ExamplePrimarySidebar />
        <ExampleSecondarySidebar />
        <ExampleContent />
      </TechDocsAddons>
    </Route>
```

 the result will be just 1 component - the custom reader.

If you do the same with: 

```tsx
<Route
      path="/docs/:namespace/:kind/:name/*"
      element={<TechDocsReaderPage />}
    >
      <TechDocsAddons>
        <ExampleHeader />
        <ExampleSubHeader />
        <ExamplePrimarySidebar />
        <ExampleSecondarySidebar />
        <ExampleContent />
      </TechDocsAddons>
       {techDocsPage}
    </Route>
```

you will get 1 component -> TechDocsAddons.

So when we were running a test for an addon, we were never able to use both a custom reader + techdocs addons because `outlet.props.children` will always just be one or another. 

If you have additional ideas on how we can accomplish supporting both I'm happy to hear. I'm also interested in what use cases you see you will use both a custom reader page and addons.

Reply:
I checkout to https://github.com/backstage/backstage/commit/71b6d3283d59e2bd4e3235289f8cd7e5a6201a3c commit and It seems to be working, logged 2 children: The custom page and the Addons wrapper

![image](https://user-images.githubusercontent.com/6290749/162450008-863892c9-bb13-480d-b9aa-3a270bfb3397.png)

I think something changed after that commit 🤔 ? And I would also recommend checking if the testing framework is using the correct reader page, it seems it is not.